core/src/main/scala/nl/vroste/zio/kinesis/client/Producer.scala

```diff
@@ -104,14 +110,66 @@ final case class ProducerSettings(
   failedDelay: Duration = 100.millis,
   metricsInterval: Duration = 30.seconds,
   updateShardInterval: Duration = 30.seconds,
-  aggregate: Boolean = false,
   allowedErrorRate: Double = 0.05,
-  shardPredictionParallelism: Int = 8
+  shardPrediction: Producer.ShardPrediction = Producer.ShardPrediction.Enabled(8),
+  aggregation: Producer.Aggregation = Producer.Aggregation.Disabled,
```

Comment:
Sadly this breaks backwards compat, but otherwise there are too many incompatible setting combinations